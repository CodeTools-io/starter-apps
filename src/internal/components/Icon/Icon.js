import React, { lazy, Suspense } from 'react';

export default function Icon({ name, ...props }) {
  const Icon = lazy(() => import(`./icons/grommet/${name}`));
  return (
    <Suspense fallback="">
      <Icon {...props} />
    </Suspense>
  );
}
