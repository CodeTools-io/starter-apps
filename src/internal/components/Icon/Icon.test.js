import React from 'react'
import Icon from './Icon'

describe('Icon', () => {
  test('can render', () => {
    const { container } = renderComponent(
      <Icon>
        <p>Content for the Icon</p>
      </Icon>
    )

    expect(container).toMatchSnapshot()
  })
})

