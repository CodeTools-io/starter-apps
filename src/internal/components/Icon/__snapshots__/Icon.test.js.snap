// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon can render 1`] = `
<div>
  
</div>
`;
